tools/benchmark.sh

```diff
@@ -134,49 +242,108 @@ const_params="
   --compression_max_dict_bytes=$compression_max_dict_bytes \
   --compression_ratio=0.5 \
   --compression_type=$compression_type \
-  --level_compaction_dynamic_level_bytes=true \
   --bytes_per_sync=$((8 * M)) \
-  --cache_index_and_filter_blocks=0 \
-  --pin_l0_filter_and_index_blocks_in_cache=1 \
+  $cache_meta_flags \
+  $o_direct_flags \
   --benchmark_write_rate_limit=$(( 1024 * 1024 * $mb_written_per_sec )) \
   \
-  --write_buffer_size=$((128 * M)) \
-  --target_file_size_base=$((128 * M)) \
-  --max_bytes_for_level_base=$((1 * G)) \
+  --write_buffer_size=$(( $write_buffer_mb * M)) \
+  --target_file_size_base=$(( $target_file_mb * M)) \
+  --max_bytes_for_level_base=$(( $l1_mb * M)) \
   \
   --verify_checksum=1 \
   --delete_obsolete_files_period_micros=$((60 * M)) \
-  --max_bytes_for_level_multiplier=8 \
+  --max_bytes_for_level_multiplier=$per_level_fanout \
   \
   --statistics=0 \
   --stats_per_interval=1 \
-  --stats_interval_seconds=60 \
+  --stats_interval_seconds=$stats_interval_seconds \
+  --report_interval_seconds=$report_interval_seconds \
   --histogram=1 \
   \
   --memtablerep=skip_list \
   --bloom_bits=10 \
   --open_files=-1 \
+  --subcompactions=$subcompactions \
   \
   $bench_args"
 
+level_const_params="
+  $const_params_base \
+  --compaction_style=0 \
+  --min_level_to_compress=$min_level_to_compress \
+  --level_compaction_dynamic_level_bytes=true \
+  --pin_l0_filter_and_index_blocks_in_cache=1 \
+  $soft_pending_arg \
+  $hard_pending_arg \
+"
+
+# TODO: these inherit level_const_params because the non-blob LSM tree uses leveled compaction
+blob_const_params="
+  $level_const_params \
+  --enable_blob_files=true \
+  --min_blob_size=$min_blob_size \
+  --blob_file_size=$blob_file_size \
+  --blob_compression_type=$blob_compression_type \
+  --enable_blob_garbage_collection=true \
+  --blob_garbage_collection_age_cutoff=$blob_gc_age_cutoff \
+  --blob_garbage_collection_force_threshold=$blob_gc_force_threshold \
+"
```

Comment:
I think we should merge and delete https://github.com/facebook/rocksdb/blob/ec0b1ff2bdabad5b024f741f68e0a6a440ea483a/tools/run_blob_bench.sh, which is duplicated.
Currently, I think it still works, but does not set the "compaction style" to "Blob" (again, I think it might be better to be consistent with Rocksdb code that not naming blobDB as a compaction style, but just "enable BlobDB". Or maybe change "compaction style" to something else).

Reply:
That is a discussion for me to have with Levy after I merge this PR. I have been waiting for ~2 months to get this one done and don't want another delay.